tests/visualization_tests/test_timeline.py

```diff
@@ -2,19 +2,26 @@
 
 import datetime
 from io import BytesIO
+from typing import Any
+from typing import Dict
+from typing import Optional
 
 import _pytest.capture
 import pytest
 
 import optuna
+from optuna.samplers._base import _CONSTRAINTS_KEY
 from optuna.study.study import Study
 from optuna.trial import TrialState
 from optuna.visualization._plotly_imports import go
 from optuna.visualization._timeline import _get_timeline_info
 from optuna.visualization._timeline import plot_timeline
 
 
-def _create_study(trial_states_list: list[TrialState]) -> Study:
+def _create_study(
+    trial_states_list: list[TrialState],
+    trial_sys_attrs: Optional[Dict[str, Any]] = None,
```

Comment:
A nit, but we use `dict` instead of `typing.Dict` as can be seen in #4508.
```suggestion
    trial_sys_attrs: dict[str, Any] | None = None,
```

Reply:
Fixed